### ISSUE TITLE
How to build Forever?

### ISSUE THREAD

**Issue by scripting** [scripting/Scripting-News#18]
I want to do a project using Forever as the starting point.

Here's the question/problem.

1. I am able to do most of what I want with [forever-monitor](https://github.com/foreverjs/forever-monitor). However I can't figure out how to ask forever-monitor for a list of the apps that are running and for the information it has about them. Here's a [screen shot](http:///scripting.com/images/2017/07/29/foreverList.png) to remind what that info is.

2. Reading the [docs](https://github.com/foreverjs/forever#foreverlist-format-callback) for forever, and the [source](https://github.com/foreverjs/forever/blob/master/lib/forever.js#L586), I see that it has a routine called `forever.list`. But when I call it from my app it returns null. I have no idea how to set things up so it returns the list. 

3. I thought of another approach, start with the Forever app, build it, and then start modifying it so it does what I want. I think that would work, but I can't figure out how to get forever.js starting with the contents of the repository. There is a forever.js file but the [package.json](https://github.com/foreverjs/forever/blob/master/package.json) file is at the top level. 

Any ideas on how to proceed? I'm happy with what I have in #1 above, but I must have the list of the apps and data about them in order for what I'm doing to work.

Dave

**Comment by scripting** [comment]
TL;DR -- I figured it out.

First a couple of observations about coding.

1. If you stare at something and can't figure out WTF is going on, put it down, come back in a few days and do it again. And again. Eventually it becomes familiar. The Nth time it just makes sense. That's what happened here.

2. Sometimes when writing up the bug report you figure out what the problem is. Specifying it to someone else helps you consider clues that you might have been rushing by or overlooking. 

That said, the answer is in the forever [docs](https://github.com/foreverjs/forever#using-in-your-code). 

![image](https://user-images.githubusercontent.com/1686843/28745865-6409a662-744f-11e7-8f6e-266043c69e2d.png)

There's a lot of vague stuff there, but you have to assume there's something real behind the admonition. I stared at it a third time, and realized what they were talking about. I just added one line of code to the routine that launches an app and It Just Worked. Here's the code with the new line emphasized.

```
function runTheApp (folder, appConfig) {
	var options = {
		max: config.maxRestarts,
		silent: config.flRunSilent,
		sourceDir: folder,
		cwd: folder,
		args: []
		};
	var child = new (foreverMonitor.Monitor) (options.script, options);
	<b>forever.startServer (child);</b>
	child.on ('exit', function () {
		console.log (folder + options.script + " has exited after " + config.maxRestarts + " restarts.");
		});
	child.start ();
	}
```
